### PR TITLE
Add quantity dropdown

### PR DESCRIPTION
**Pull request by AntonoffEvgeniy**

### CODE REVIEW

File: ShopApp/App/Extensions/UIViewController+Alert.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//
+//  Created by Evgeniy Antonov on 4/23/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import UIKit
+
+private let kQuantityCountMax = 3
+
+extension UIViewController {
```
Comment (plzen) [inline ShopApp/App/Extensions/UIViewController+Alert.swift]: Is it possible to write tests for this extension?

File: ShopApp/App/Extensions/UIViewController+Alert.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        let cancelAction = UIAlertAction(title: "Button.Cancel".localizable, style: .cancel)
+        
+        alertController.addAction(doneAction)
+        alertController.addAction(cancelAction)
+        
+        present(alertController, animated: true)
+    }
+}
+
+extension UIAlertController {
```
Comment (plzen) [inline ShopApp/App/Extensions/UIViewController+Alert.swift]: Is it possible to write tests for this extension?

File: ShopApp/App/Modules/Main/Controllers/ProductDetailsViewController.swift [hunk 3]
```diff
@@ -280,4 +281,16 @@ class ProductDetailsViewController: BaseViewController<ProductDetailsViewModel>,
         productDetailsViewController.productId = product.id
         navigationController.pushViewController(productDetailsViewController, animated: true)
     }
+    
+    // MARK: - QuantityDropDownViewDelegate
+    
+    func quantityDropDownView(_ view: QuantityDropDownView, didSelectMoreWith quantity: Int) {
+        showQuantityAlert(with: quantity) { [weak self] text in
```
Comment (plzen) [inline ShopApp/App/Modules/Main/Controllers/ProductDetailsViewController.swift]: Add tests for this method

File: ShopApp/App/Modules/Cart/Controllers/CartViewController.swift [hunk 4]
```diff
@@ -102,6 +102,15 @@ class CartViewController: BaseViewController<CartViewModel>, CartEmptyDataViewDe
         viewModel.update(cartProduct: cartProduct, quantity: quantity)
     }
     
+    func tableViewCell(_ tableViewCell: CartTableViewCell, didSelectMoreFor cartProduct: CartProduct, with quantity: Int) {
+        showQuantityAlert(with: quantity) { [weak self] text in
```
Comment (plzen) [inline ShopApp/App/Modules/Cart/Controllers/CartViewController.swift]: Add tests for this method

File: ShopApp/App/Modules/Shared/Views/Cells/TableView/QuantityDropDownTableViewCell/QuantityDropDownTableViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import DropDown
+
+class QuantityDropDownTableViewCell: DropDownCell {
+    private let fontSize: CGFloat = 14
+    
+    public func configure(with textCenter: Bool) {
+        optionLabel.textAlignment = textCenter ? .center : .left
```
Comment (plzen) [inline ShopApp/App/Modules/Shared/Views/Cells/TableView/QuantityDropDownTableViewCell/QuantityDropDownTableViewCell.swift]: Add tests for this block

File: ShopApp/App/Modules/Shared/Views/QuantityDropDownView/QuantityDropDownView.swift [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        var quantity = (sender.text as NSString?)?.integerValue ?? 0
+        if quantity < cartProductQuantityMin {
+            quantity = cartProductQuantityMin
+            textField.text = String(quantity)
+        }
+        delegate?.quantityDropDownView?(self, didChange: quantity)
+    }
+    
+    private func showDropDown() {
+        dropDown.dataSource = DropDownItem.allValues
```
Comment (plzen) [inline ShopApp/App/Modules/Shared/Views/QuantityDropDownView/QuantityDropDownView.swift]: Can we cover also this code with tests?

File: ShopApp/App/Extensions/UIViewController+Alert.swift (hunk 1 above)
Comment (AntonoffEvgeniy) [reply ShopApp/App/Extensions/UIViewController+Alert.swift]: We can't get presented alert controller in tests

File: ShopApp/App/Modules/Main/Controllers/ProductDetailsViewController.swift (hunk 3 above)
Comment (AntonoffEvgeniy) [reply ShopApp/App/Modules/Main/Controllers/ProductDetailsViewController.swift]: We can't check if alert controller was presented

File: ShopApp/App/Modules/Shared/Views/Cells/TableView/QuantityDropDownTableViewCell/QuantityDropDownTableViewCell.swift (hunk 5 above)
Comment (AntonoffEvgeniy) [reply ShopApp/App/Modules/Shared/Views/Cells/TableView/QuantityDropDownTableViewCell/QuantityDropDownTableViewCell.swift]: It's subclass of pod's class. I can't find way to generate this cell for tests

File: ShopApp/App/Modules/Shared/Views/QuantityDropDownView/QuantityDropDownView.swift (hunk 6 above)
Comment (AntonoffEvgeniy) [reply ShopApp/App/Modules/Shared/Views/QuantityDropDownView/QuantityDropDownView.swift]: dropDown is a private property

**Pull request merged**